### DISCUSSION TITLE
Is it safe to update from release to CI docker image

### DISCUSSION THREAD

**Discussion by rehanone** [simple-login/app#1026] [Self-hosting question]
We have many versions tagged in the past few weeks. None of these are pushed to release images on docker hub where 3.4.0 is still the latest. I can see that new tags are pushed to the CI branch. I am wondering if it is safe to update from release 3.4.0 to CI 4.5.2?

**Comment by 0xspade** [comment]
They say that the update for self-hosted will be May or June with no exact dates, but I hope they give us a specific date, We are just waiting like you.

**Comment by AzuritBadger** [comment]
I was wondering the same thing. It has been been a thorn in my side for quite some time knowing that one of the services I host wasn't updated in over a year. Today I tried setting up the latest tag of app-ci, which worked perfectly fine with the config i had with 3.4.0. Only difference was I needed to use `alembic upgrade head` instead of `flask db upgrade`. Considering that their own production runs of the latest versioned tag (can be seen by comparing the git commit hash from https://app.simplelogin.io/git with the one from the GitHub release), I think it is safe to use (of course a backup should always be made). Since my test migration went well, I'll probably will switch my production instance this evening.

**Reply by nguyenkims** [reply]
Yes apart from the `alembic upgrade head` instead of `flask db upgrade` there shouldn't be no issue upgrading to the latest version.

Actually it'd help us a lot if you can try to upgrade your current version to the latest one and let us know if any error happens. You can find the latest docker image on `simplelogin/app-ci:latest` image.

**Reply by AzuritBadger** [reply]
Sure! Upgrading went without any issue, I was actually done in under 10 minutes :). The only notable log messages I encountered where entries about unreachable symbols and that the `MAX_NB_EMAIL_FREE_PLAN` variable isn't set. 

```
MAX_NB_EMAIL_FREE_PLAN is not set, use 5 as default value
Paddle param not set
Upload files to local dir


Symbol 'mailbox_or_url_list' is unreachable
Symbol 'delim' is unreachable
Symbol 'mailbox_or_url' is unreachable
Symbol 'url' is unreachable
Symbol 'mailbox_or_url_list' is unreachable
Symbol 'delim' is unreachable
Symbol 'mailbox_or_url' is unreachable
Symbol 'mailbox' is unreachable
Symbol 'url' is unreachable
Symbol 'angle_addr' is unreachable
Symbol 'name_addr' is unreachable
Symbol 'phrase' is unreachable
Symbol 'mailbox_or_url_list' is unreachable
Symbol 'delim' is unreachable
Symbol 'mailbox_or_url' is unreachable
Symbol 'mailbox' is unreachable
Symbol 'addr_spec' is unreachable
Symbol 'angle_addr' is unreachable
Symbol 'name_addr' is unreachable
Symbol 'phrase' is unreachable
Symbol 'local_part' is unreachable
Symbol 'domain' is unreachable
Symbol 'quoted_string' is unreachable
Symbol 'domain_literal' is unreachable
Symbol 'quoted_string_text' is unreachable
Symbol 'domain_literal_text' is unreachable
Symbol 'mailbox_or_url_list' is unreachable
Symbol 'delim' is unreachable
```

I also spun up a completely fresh instance with the latest app-ci release tag, which yielded the same messages. Should this be a concern and if yes, where should these symbols be set, and  what are they used for? Using the latest Postgres tag (14.3) also worked for the fresh installation, but when trying to upgrade from 12.1 to 14.3, the container was stuck with status `Restarting (1)`, so I kept 12.1 for my production instance. I think this is probably a Postgres thing and not related to SimpleLogin. 

The only difference to app.simplelogin.io was that none of my instances had the "Subdomains" tab. I had a quick look at the source code, and from what I can tell subdomains need to be enabled on a per user basis?

All in all everything works as it should. I would also like too use this opportunity to say thank you for your awesome work, I cannot understate how much I love this software, it has been such a game changer for me :)

**Reply by 0xspade** [reply]
Looking forward to this one. There should be an update.

**Reply by Yougoshatenshi** [reply]
I also upgraded from 3.4.0 to app-ci.
Beside the unreachable symbols my dashboard shows duplicated aliases.

I have 62 aliases and it shows 197 entries of them, with one alias has up to 50 entries, maybe it has to do with the forwarded amount.
I might test it later again from a backup when i got the time for it.

Otherwise i don't have any problems.

**Reply by AzuritBadger** [reply]
I didn't even notice, but I just checked it, there seem to be duplicates for me too, but not for every alias. Some are doubled, and I spotted one that exists 5 times now, and interestingly has 5 forwards. On the other hand, an alias with 335 forwards only appears once. 

Edit: I just found one with over 50 duplicates, and sending an email to said alias removed all duplicates.

**Comment by Metabaron1** [comment]
@nguyenkims If you add an arm64 image build to your CI, I will be happy to help testing the upgrade from 3.4.0 :)

**Comment by 0xspade** [comment]
@nguyenkims where update? you said it's May-June.
